### DISCUSSION TITLE
How to obtain XYZ from the RGB value for the display?

### DISCUSSION THREAD

**Discussion by chesschi** [colour-science/colour#730] [Q&A] [answered]
In the Annex 3 of [Colour conversion from Recommendation ITU-R BT.709 to Recommendation ITU-R BT.2020](https://www.itu.int/dms_pubrec/itu-r/rec/bt/R-REC-BT.2087-0-201510-I!!PDF-E.pdf), there is an example how you can convert the 10-bit RGB to xyY. How can I use the colour library to calculate the xyY value like the example?

Quoted from the example:
> The red object is selected to be at 20 cd/m2 luminance and the same chromaticity as the Rec. 709 red primary. This can be expressed in Yxy coordinates as Y = 20, x = 0.64, y = 0.33. If the Rec. 709 camera is assumed to have a sensor which utilizes perfect CIE1931 colour matching functions, and the iris is adjusted so the red object produces a normalized Y output from the sensor of 0.2, the result is a 10-bit Rec. 709 encoded R’G’B’ output of R’ = 914, G’ = 64, B’ = 64. After being decoded by the Rec. 709 display, the result is an output of Y = 19.8, x = 0.640, y = 0.330 which is very close to the original scene colour.

I have followed the [example](https://github.com/colour-science/colour/issues/427) as per @nick-shaw:
`colour.XYZ_to_xyY(colour.utilities.dot_vector(colour.models.BT2020_COLOURSPACE.RGB_to_XYZ_matrix, colour.models.eotf_ST2084(np.array([914, 64, 64]) / 1023)) / 10000)`
and this will output [0.70796336, 0.29200343, 0.09656418] which is not close to [0.64, 0.33, 19.8?] at all, although I don't understand why they express Y as non-normalized value.

It would be great if you can shed some light on this. Many thanks!

**Comment by thomasmansencal** [comment, marked as answer]
Hi @chesschi,

You could compute them as follows, there is the hidden assumption which is that the values are encoded in legal range:

```python
import numpy as np
import colour

BT709 = colour.RGB_COLOURSPACES['ITU-R BT.709']

xyY = np.array([0.64, 0.33, 20]) / [1, 1, 100]
RGB = colour.XYZ_to_RGB(
    colour.xyY_to_XYZ(xyY),
    BT709.whitepoint,
    BT709.whitepoint,
    BT709.XYZ_to_RGB_matrix)
RGB_p = colour.oetf(RGB, method='ITU-R BT.709')
print(colour.full_to_legal(RGB_p, bit_depth=10, out_int=True))

# array([914,  64,  64])
```

I hope this helps! :)

**Comment by thomasmansencal** [comment]
Closing this one @chesschi, feel free to add any comments!
